Sources/Commands/SwiftBuildTool.swift

```diff
@@ -157,6 +162,29 @@ public struct SwiftBuildTool {
             }
         
             switch mode {
+            case .repl(let toolchain):
+                let (rootPackage, externalPackages) = try fetch(opts.path.root)
+                let (modules, externalModules, _) = try transmute(rootPackage, externalPackages: externalPackages)
+
+                /// TODO: Need to create DSOs for C executables as well.
+                let products: [Product] = modules.flatMap{ $0 as? SwiftModule }
+                                                 .map{ Product(name: $0.name, type: .Library(.Dynamic), modules: [$0]) }
+
+                let yaml = try describe(opts, .debug, modules, Set(externalModules), products, toolchain: toolchain)
```

Comment:
If this mode requires us to build additional stuff, I would prefer that be modeled as an extra argument to the describe() function selecting the mode, and then have that function embed the implementation details of what extra things need to be done to support that mode (i.e., the computation of other products)